object/saml_idp.go

```diff
@@ -103,9 +104,20 @@ func NewSamlResponse(user *User, host string, certificate string, destination st
 	displayName.CreateAttr("NameFormat", "urn:oasis:names:tc:SAML:2.0:attrname-format:basic")
 	displayName.CreateElement("saml:AttributeValue").CreateAttr("xsi:type", "xs:string").Element().SetText(user.DisplayName)
 
-	roles := attributes.CreateElement("saml:Attribute")
-	roles.CreateAttr("Name", "Roles")
-	roles.CreateAttr("NameFormat", "urn:oasis:names:tc:SAML:2.0:attrname-format:basic")
+	var roles *etree.Element
+	if strings.Contains(destination, "tencent") && application.SamlAttribute != nil {
+		for _, item := range application.SamlAttribute {
+			roles = attributes.CreateElement("saml:Attribute")
+			roles.CreateAttr("Name", item.AttributeName)
+			roles.CreateAttr("NameFormat", item.Nameformat)
+			roles.CreateAttr("FriendlyName", "tencent cloud")
+			roles.CreateElement("saml:AttributeValue").CreateAttr("xsi:type", "xs:string").Element().SetText(item.Value)
```

Comment:
Does `SamlAttributes` only used for SAML roles? If yes, why not rename to `SamlRoles`?

Reply:
This attribute can include the user's name, email, role, department, identity, permissions, or any other information related to the user's identity.